packages/rocketchat-videobridge/client/public/external_api.js

```diff
@@ -0,0 +1,377 @@
+        if(!this.eventHandlers[event])
+        {
+            console.error("The event " + event + " is not registered.");
+            return;
+        }
+        var message = {type: "event", action: "remove", events: [event]};
+        delete this.eventHandlers[event];
+        this.sendMessage(message);
+    };
+
+    /**
+     * Removes event listeners.
+     * @param events array with the names of the events.
+     */
+    JitsiMeetExternalAPI.prototype.removeEventListeners
+        = function (events) {
+        var eventsArray = [];
+        for(var i = 0; i < events.length; i++)
+        {
+            var event = events[i];
+            if(!this.eventHandlers[event])
+            {
+                console.error("The event " + event + " is not registered.");
+                continue;
+            }
+            delete this.eventHandlers[event];
+            eventsArray.push(event);
+        }
+
+        if(eventsArray.length > 0)
+        {
+            this.sendMessage(
+                {type: "event", action: "remove", events: eventsArray});
+        }
+
+    };
+
+    /**
+     * Processes message events sent from Jitsi Meet
+     * @param event the event
+     */
+    JitsiMeetExternalAPI.prototype.processMessage = function(event) {
+        var message;
+        try {
+            message = JSON.parse(event.data);
+        } catch (e) {}
+
+        if(!message.type) {
+            console.error("Message without type is received.");
+            return;
+        }
+        switch (message.type) {
+            case "system":
+                if(message.loaded) {
+                    this.onFrameLoaded();
+                }
+                break;
+            case "event":
+                if(message.action != "result" ||
+                    !message.event || !this.eventHandlers[message.event]) {
+                    console.warn("The received event cannot be parsed.");
+                    return;
+                }
+                this.eventHandlers[message.event](message.result);
+                break;
+            default :
+                console.error("Unknown message type.");
+                return;
+        }
+    };
+
+    /**
+     * That method is called when the Jitsi Meet is loaded. Executes saved
+     * commands that are send before the frame was loaded.
+     */
+    JitsiMeetExternalAPI.prototype.onFrameLoaded = function () {
+        this.frameLoaded = true;
+        for (var i = 0; i < this.initialCommands.length; i++) {
+            this.sendMessage(this.initialCommands[i]);
+        }
+        this.initialCommands = null;
+    };
+
+    /**
+     * Setups the listener for message events from Jitsi Meet.
+     */
+    JitsiMeetExternalAPI.prototype.initListeners = function () {
+        var self = this;
+        this.eventListener = function (event) {
+            self.processMessage(event);
+        };
+        if (window.addEventListener) {
+            window.addEventListener('message',
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2736174701/issues/source?bid=3199634&fileBranchId=3342079#l345)